src/classes/COUR_CannotDelete_TDTM.cls

```diff
@@ -0,0 +1,87 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Course
+* @group-content ../../ApexDocContent/Course.htm
+* @description Stops a Course from being deleted if it has any Course Offering or Plan Requirement child records.
+*/
+public with sharing class COUR_CannotDelete_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description Get the setting of preventing Course deletion
+    */
+    private static Boolean enabledPreventCourseDeletion = UTIL_CustomSettingsFacade.getSettings().Prevent_Course_Deletion__c;
+    
+    /*******************************************************************************************************
+    * @description Stops a Course from being deleted if it has any Course Offering 
+    * or Plan Requirement child records.
+    * @param listNew the list of Courses from trigger new.
+    * @param listOld the list of Courses from trigger old.
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @param objResult the describe for Courses
+    * @return dmlWrapper.
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
+    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+		
+        if (!COUR_CannotDelete_TDTM.enabledPreventCourseDeletion) {
+            return new DmlWrapper(); 
+        }
+        
+      	Map<Id, Course__c> oldMap = new Map<Id, Course__c>((List<Course__c>)oldList);
+      
+        if (triggerAction == TDTM_Runnable.Action.BeforeDelete) {
+            for (Course__c course : [SELECT ID,
+                            (SELECT ID FROM Course__c.Course_Offerings__r LIMIT 1),
+                     		(SELECT ID FROM Course__c.Plan_Requirements__r LIMIT 1)
+                            FROM Course__c
+                            WHERE ID IN :oldlist])
+            {
+                if (this.hasChildRecords(course)) {
+                    Course__c courseInContext = oldmap.get(course.ID);
+                    courseInContext.addError(Label.CannotDelete);
+                }
+            }
+        }
+
+      return new DmlWrapper();
+   }
+
+   /*******************************************************************************************************
+   * @description Evaluates whether the Course has any child related records.
+   * @param course is the current Course record.
+   * @return Boolean.
+   ********************************************************************************************************/
+   @testVisible
+   private Boolean hasChildRecords(Course__c course) {
+   		return (!course.Course_Offerings__r.isEmpty() || !course.Plan_Requirements__r.isEmpty());
+   }
```

Comment:
@kdy618 Please resolve the spacing issues on this to conform to the four spaces for indentation standard.